control_plane/src/endpoint.rs

```diff
@@ -183,6 +170,30 @@ impl ComputeControlPlane {
 
         Ok(ep)
     }
+
+    pub fn check_conflicting_endpoints(
+        &self,
+        mode: ComputeMode,
+        tenant_id: TenantId,
+        timeline_id: TimelineId,
+    ) -> Result<()> {
+        if matches!(mode, ComputeMode::Primary) {
+            // this check is not complete, as you could have a concurrent attempt at
+            // creating another primary, both reading the state before checking it here,
+            // but it's better than nothing.
+            let mut duplicates = self.endpoints.iter().filter(|(_k, v)| {
+                v.tenant_id == tenant_id
+                    && v.timeline_id == timeline_id
+                    && v.mode == mode
+                    && v.status().starts_with("running")
```

Comment:
I was expecting #5854 to come up as erroring here, but did not.

Reply:
TODO(me): check if this status is trustworthy, I don't know what it is.

Reply:
Ok, it seems trustworthy, it will try to connect if the pidfile exists. I am surprised how this does not become slow, but still this is more asserts than before so I am happy about this.